mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -1982,14 +1983,90 @@ struct FoldDimOfCollapseShape : public OpRewritePattern<DimOp> {
     return success();
   }
 };
+
+struct ConvertToStaticExpandShape : public OpRewritePattern<ExpandShapeOp> {
```

Comment:
Can you please document the pattern with a high-level description of what this pattern is doing? That'll be useful to future folks having to debug or improve this :)
(or just skimming the codebase).

Thanks.